code/modules/atmospherics/machinery/air_alarm/air_alarm_interact.dm

```diff
@@ -92,10 +92,12 @@
 /obj/machinery/airalarm/deconstruct(disassembled = TRUE)
 	if(!(flags_1 & NODECONSTRUCT_1))
 		new /obj/item/stack/sheet/iron(loc, 2)
-		var/obj/item/I = new /obj/item/electronics/airalarm(loc)
-		if(!disassembled)
-			I.take_damage(I.max_integrity * 0.5, sound_effect=FALSE)
-		new /obj/item/stack/cable_coil(loc, 3)
+		if((buildstage & AIR_ALARM_BUILD_NO_WIRES) || (buildstage & AIR_ALARM_BUILD_COMPLETE))
+			var/obj/item/I = new /obj/item/electronics/airalarm(loc)
+			if(!disassembled)
+				I.take_damage(I.max_integrity * 0.5, sound_effect = FALSE)
```

Comment:
```suggestion
			var/obj/item/alarm = new /obj/item/electronics/airalarm(loc)
			if(!disassembled)
				alarm.take_damage(alarm.max_integrity * 0.5, sound_effect = FALSE)
```

may as well clean up this single-letter variable while in the area

Reply:
also we could also do something like

```dm
var/obj/item/electronics/airalarm/alarm = new(loc)
```

but that's just shifting the words to a different spot in the line and isn't _truly_ necessary